fe/fe-core/src/main/java/org/apache/doris/planner/HudiScanNode.java

```diff
@@ -229,41 +225,11 @@ private void initParams(Analyzer analyzer) {
         context.slotDescByName = slotDescByName;
     }
 
-
-    /**
-     * Extracts partition predicate from SelectStmt.whereClause that can be pushed down to Hive.
-     */
-    private void extractHivePartitionPredicate() throws DdlException {
-        ListIterator<Expr> it = conjuncts.listIterator();
-        while (it.hasNext()) {
-            ExprNodeGenericFuncDesc hiveExpr = HiveMetaStoreClientHelper.convertToHivePartitionExpr(
-                    it.next(), partitionKeys, hudiTable.getName());
-            if (hiveExpr != null) {
-                hivePredicates.add(hiveExpr);
-            }
-        }
-        int count = hivePredicates.size();
-        // combine all predicate by `and`
-        // compoundExprs must have at least 2 predicates
-        if (count >= 2) {
-            hivePartitionPredicate = HiveMetaStoreClientHelper.getCompoundExpr(hivePredicates, "and");
-        } else if (count == 1) {
-            // only one predicate
-            hivePartitionPredicate = (ExprNodeGenericFuncDesc) hivePredicates.get(0);
-        } else {
-            // have no predicate, make a dummy predicate "1=1" to get all partitions
-            HiveMetaStoreClientHelper.ExprBuilder exprBuilder =
-                    new HiveMetaStoreClientHelper.ExprBuilder(hudiTable.getName());
-            hivePartitionPredicate = exprBuilder.val(TypeInfoFactory.intTypeInfo, 1)
-                    .val(TypeInfoFactory.intTypeInfo, 1)
-                    .pred("=", 2).build();
-        }
-    }
-
     private InputSplit[] getSplits() throws UserException, IOException {
         String splitsPath = basePath;
         if (partitionKeys.size() > 0) {
-            extractHivePartitionPredicate();
+            hivePartitionPredicate = HiveMetaStoreClientHelper.convertToHivePartitionExpr(
```

Comment:
We call `convertToHivePartitionExpr` twice in `HudiScanNode`.
I think the call in `getFileStatus()` can be removed?